code/hud/hudets.cpp

```diff
@@ -202,37 +202,44 @@ void ai_manage_ets(object* obj)
 {
 	ship* ship_p = &Ships[obj->instance];
 	ship_info* ship_info_p = &Ship_info[ship_p->ship_info_index];
+	ai_info	*aip = &Ai_info[Ships[Pl_objp->instance].ai_index];
 
 	if ( ship_info_p->power_output == 0 )
 		return;
 
 	if (ship_p->flags[Ship::Ship_Flags::Dying])
 		return;
 
-	// check if any of the three systems are not being used.  If so, don't allow energy management.
-	if ( !ship_p->ship_max_shield_strength || !ship_info_p->max_speed || !ship_info_p->max_weapon_reserve)
-		return;
+	// check if weapons or engines are not being used. If so, don't allow energy management.
+	// also check if the ship has no shields and if the AI is allowed to manage weapons and engines --wookieejedi
+	if ( !ship_info_p->max_speed || !ship_info_p->max_weapon_reserve || 
+		(!ship_p->ship_max_shield_strength && !aip->ai_profile_flags[AI::Profile_Flags::nonshielded_ships_can_manage_ets]) ) {
+			return;
+	}
 
-	float shield_left_percent = get_shield_pct(obj);
 	float weapon_left_percent = ship_p->weapon_energy/ship_info_p->max_weapon_reserve;
 
-	// maximum level check
+	// maximum level check for weapons
 	//	MK, changed these, might as well let them go up to 100% if nothing else needs the recharge ability.
 	if ( weapon_left_percent == 1.0f) {
 		decrease_recharge_rate(obj, WEAPONS);
 	}
 
-	if (!(obj->flags[Object::Object_Flags::No_shields]) && (shield_left_percent == 1.0f)) {
-		decrease_recharge_rate(obj, SHIELDS);
-	}
-
-	// minimum check
-
-	if (!(obj->flags[Object::Object_Flags::No_shields]) && (shield_left_percent < SHIELDS_MIN_LEVEL_PERCENT)) {
-		if ( weapon_left_percent > WEAPONS_MIN_LEVEL_PERCENT )
-			increase_recharge_rate(obj, SHIELDS);
+	bool has_shields = !ship_p->ship_max_shield_strength && !(obj->flags[Object::Object_Flags::No_shields]);
```

Comment:
Is this condition correct?  You're checking to see whether the ship has 0% maximum shields and also doesn't have the no-shields flag.

Reply:
Correct, though with Cyborg's comments this has changed.